### PR TITLE
Feature/#1 Transfer widget modal 

### PR DESCRIPTION
**Pull request by BeneRichi**
# Pull Request: Feature/#1 Transfer widget modal

## Description
This pull request adds a new "Transfer Widget Modal" feature to the application. The feature allows users to view their transaction details inside a simple and clear modal with basic informations about their transaction. 

The following changes have been made:
- Build for NFT transfers and Token transfers (More activites will come like : Swap)
- Removed a-tag around ActivityItem in Activity.tsx 
- Created ActivityDetails.tsx modal component with two templates, one for NFT transaction and one for Token transaction
- Save current transaction in state named "selectedTransaction"
- Open modal on click of ActivityItem.tsx 
- Pass necessary props into ActivityDetails.tsx of the selectedTransaction
- Style component in activity-details.css

## Tasks

- [x] Test the feature on different browsers

## Thank you!

Thanks for taking the time to review this pull request. I'm looking forward to your feedback!

### CODE REVIEW

**Comment by BeneRichi** [comment]
@matjazonline Thanks for taking the time to review this pull request.

File: src/pages/dashboard/Activity/Activity.tsx [hunk 1]
```diff
@@ -14,53 +15,77 @@ noActivityTokenDisplay.name = 'No account history yet.';
 
 export const Activity = (): JSX.Element => {
   const transfers = hooks.useObservableState(appState.transferHistory$);
-
+  const [isActivityModalOpen, setActivityModalOpen] = useState(false);
   const signer: ReefSigner | undefined |null = hooks.useObservableState(appState.selectedSigner$);
   const network: Network|undefined = hooks.useObservableState(appState.currentNetwork$);
+  const [selectedTransaction, setTransaction] = useState<TokenTransfer|null>(null);
```
Comment (matjazonline) [inline src/pages/dashboard/Activity/Activity.tsx:21]: should be setSelectedTransaction

File: src/pages/dashboard/Activity/Activity.tsx [hunk 2]
```diff
@@ -14,53 +15,77 @@ noActivityTokenDisplay.name = 'No account history yet.';
...
+                    )
+                }
       </div>
 
       <div className={`col-12 card  ${transfers?.length ? 'card-bg-light' : ''}`}>
         {!!transfers && !transfers.length && <div className="no-token-activity">No recent transfer activity.</div>}
         {!!transfers && !!transfers.length && (
         <div>
-            {transfers.slice(0, 10).map((item, index) => (
+          {transfers.slice(0, 10).map((item, index) => (
```
Comment (matjazonline) [inline src/pages/dashboard/Activity/Activity.tsx:49]: why take 10 - better if query sets nr of items

**Review by matjazonline — CHANGES_REQUESTED** [review]

**Comment by BeneRichi** [comment]
@matjazonline Can you explain the reason why there are accounts where a match cannot be found in the signer array, resulting in the inability to retrieve a name for the wallet? I have used "no name" as a fallback solution in these cases.

**Comment by matjazonline** [comment]
signers array is from accounts in extension and this can occur if account is not present

**Comment by BeneRichi** [comment]
Got it, make sense. Thanks for clarification.

**Comment by BeneRichi** [comment]
@matjazonline PR is ready for review.

**Pull request closed without merging**